src/mod/endpoints/mod_sofia/sofia_presence.c

```diff
@@ -2987,12 +2987,29 @@ static int sofia_presence_sub_callback(void *pArg, int argc, char **argv, char *
 
 			if (!strcasecmp(astate, "early") || !strcasecmp(astate, "confirmed")) {
 
+				char *cmd;
+				char *callee_id_name = NULL;
+				switch_stream_handle_t user_data_stream = { 0 };
+
 				clean_to_user = switch_mprintf("%s", sub_to_user ? sub_to_user : to_user);
 				clean_from_user = switch_mprintf("%s", from_id ? from_id : from_user);
 
+				SWITCH_STANDARD_STREAM(user_data_stream);
+				cmd = switch_mprintf("%s@%s var effective_caller_id_name", clean_to_user, host);
```

Comment:
Where is `cmd` freed?

Reply:
When it falls out of scope?  cmd is declared on the stack at 2990

Reply:
is memory allocated by `switch_mprintf` freed?

Reply:
Ack